dspace-api/src/main/java/org/dspace/scripts/DSpaceRunnable.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.scripts;
+
+import java.sql.SQLException;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.core.Context;
+import org.dspace.scripts.handler.DSpaceRunnableHandler;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Required;
+
+/**
+ * This abstract class is the class that should be extended by each script.
+ * it provides the basic variables to be hold by the script as well as the means to initialize, parse and run the script
```

Comment:
Do all `DSpaceRunnable` scripts also need to be added to `scripts.xml` as a bean?  If so, we should add a note documenting that to this abstract class.